File: src/atr_pkg/scripts/ma_node_1.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry
#define callback function
def chao_callback(msg):
    rospy.logwarn(msg.grade)
    rospy.logwarn(msg.star)
    rospy.loginfo(msg.data)

def yao_callback(msg):
    rospy.loginfo(msg.data)

if __name__=="__main__":
    try:
        #init node
        rospy.init_node("ma_node")
        rospy.logwarn("ma_node is started  ")

        #subscribe topic
        sub=rospy.Subscriber("chao_node_toptic",Carry,chao_callback,queue_size=10)
        sub_1=rospy.Subscriber("yao_node_topic",String,yao_callback,queue_size=10)

        rospy.spin()#查看消息是否到达
    except rospy.ROSInterruptException:
        rospy.logerr("节点运行中断")

File: src/atr_pkg/src/ma_node_1.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>
#include<qq_msgs/Carry.h>
void chao_callback(qq_msgs::Carry msg){
    // printf(msg.data.c_str());
    // printf("\n");
    ROS_WARN(msg.grade.c_str());
    ROS_WARN("星级：%d",msg.star);
    ROS_INFO(msg.data.c_str());
}
void yao_callback(std_msgs::String msg){
    //ROS_INFO(msg.data.c_str());
    ROS_WARN(msg.data.c_str());
}
int main(int argc,char  *argv[])
{
    setlocale(LC_ALL,"");//正常显示中文
    ros::init(argc,argv,"ma");
    printf( "小马哥\n");
    ros::NodeHandle nh;
    ros::Subscriber sub=nh.subscribe("qiu_dai_fei",10,yao_callback);
    ros::Subscriber sub_2=nh.subscribe("kuaishangchekaiheiqun",10,chao_callback);
    //ros::Rate loop_rate(100);
    while (ros::ok())
    {
          ros::spinOnce();
          //ros::spin();//两个都能用
        //loop_rate.sleep();
    }
    
    return 0;
}